src/app/app.component.ts

```diff
@@ -129,20 +136,18 @@ export class AppComponent implements OnInit, OnDestroy {
       // only include episodes with publish dates
         .filter(doc => doc['publishedAt'])
         .map(doc => {
-        return {
-          doc,
-          id: doc['id'],
-          seriesId: podcast.seriesId,
-          title: doc['title'],
-          publishedAt: doc['publishedAt'] ? new Date(doc['publishedAt']) : null
-        };
-      });
-      episodes.forEach((e) => {
-        distObsv.push(this.getEpisodePodcastDistribution(e));
-      });
+          return {
+            doc,
+            id: doc['id'],
+            seriesId: podcast.seriesId,
+            title: doc['title'],
+            publishedAt: doc['publishedAt'] ? new Date(doc['publishedAt']) : null
+          };
+        });
+      const distros$ = episodes.map((e) => this.getEpisodePodcastDistribution(e));
       // wait for all the episode podcast distributions, then add all episodes to state at once
-      Observable.zip(...distObsv).subscribe(() => {
-        this.store.dispatch(new CmsAllPodcastEpisodeGuidsAction({podcast, episodes}));
+      Observable.zip(...distros$).subscribe(() => {
+        this.store.dispatch(new CmsAllPodcastEpisodeGuidsAction({podcast, episodes: episodes.filter(e => e.guid)}));
```

Comment:
(copy/paste of same podcast comment) I either saw a podcast without a feederId or an episode without a guid, can't remember for sure. I'm having a hard time remembering, but an issue came up about the series or episode right around the same time. Figured it was better safe than sorry to do metrics for the good data and just skip over data problems it can't handle.